src/ProxyManager/ProxyGenerator/Util/PublicScopeSimulator.php

```diff
@@ -73,23 +84,29 @@ public static function getPublicAccessSimulationCode(
     /**
      * This will generate code that triggers a notice if access is attempted on a non-existing property
      */
-    private static function getUndefinedPropertyNotice(string $operationType, string $nameParameter): string
+    private static function getUndefinedPropertyNotice(string $operationType, string $nameParameter, ?string $interfaceName = null): string
     {
         if ($operationType !== self::OPERATION_GET) {
             return '';
         }
 
-        return '    $backtrace = debug_backtrace(false, 1);' . "\n"
+        $code = '    $backtrace = debug_backtrace(false, 1);' . "\n"
             . '    trigger_error(' . "\n"
             . '        sprintf(' . "\n"
             . '            \'Undefined property: %s::$%s in %s on line %s\',' . "\n"
-            . '            get_parent_class($this),' . "\n"
+            . '            ' . ($interfaceName !== null ? var_export($interfaceName, true) : 'get_parent_class($this)') . ',' . "\n"
```

Comment:
Ah, wait, so this just produces a better notice for the interface?

Reply:
this removes a fatal error (see my other comment) and turns it into a proper notice (which also tells "don't do this!", but in a better way that a fatal error)